flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableConfig.java

```diff
@@ -343,4 +356,10 @@ public void addJobParameter(String key, String value) {
     public static TableConfig getDefault() {
         return new TableConfig();
     }
+
+    public static TableConfig getBatchTableConfig() {
```

Comment:
We are introducing a lot of public API methods for which we need to provide backwards compatibility across releases. This can simply be done in a private method in `BatchTableEnvironment`.